src/mavlink/manager.rs

```diff
@@ -1 +1,203 @@
+                address,
+                connection,
+                sender,
+            })),
+            ids: Arc::new(Mutex::new(vec![])),
+        };
+
+        let connection = this.connection.clone();
+        std::thread::Builder::new()
+            .name("MavSender".into())
+            .spawn(move || Manager::sender_loop(connection))
+            .expect("Failed to spawn MavSender thread");
+
+        let connection = this.connection.clone();
+        std::thread::Builder::new()
+            .name("MavReceiver".into())
+            .spawn(move || Manager::receiver_loop(connection))
+            .expect("Failed to spawn MavReceiver thread");
+
+        this
+    }
+}
+
+impl Manager {
+    // Construct our manager, should be done inside main
+    #[instrument(level = "debug")]
+    pub fn init() {
+        MANAGER.as_ref();
+    }
+
+    #[instrument(level = "debug", skip(inner))]
+    fn receiver_loop(inner: Arc<Mutex<Connection>>) {
+        loop {
+            std::thread::sleep(std::time::Duration::from_micros(10));
+
+            // Receive from the Mavlink network
+            let (header, message) = match inner.lock().unwrap().connection.recv() {
+                Ok(message) => message,
+                Err(error) => {
+                    if let mavlink::error::MessageReadError::Io(io_error) = &error {
+                        if io_error.kind() == std::io::ErrorKind::WouldBlock {
+                            trace!("Blocked I/O, probably disconnected!");
+                            continue;
+                        }
+                    }
+                    error!("Failed receiving message from Mavlink Connection: {error:?}");
+                    break;
+                }
+            };
+
+            trace!("Message received: {header:?}, {message:?}");
+
+            // Send the received message to the cameras
+            if let Err(error) = inner
+                .lock()
+                .unwrap()
+                .sender
+                .send(Message::Received((header, message)))
+            {
+                error!("Failed handling message: {error:?}");
+                continue;
+            }
+        }
+    }
+
+    #[instrument(level = "debug", skip(inner))]
+    fn sender_loop(inner: Arc<Mutex<Connection>>) {
+        let mut receiver = { inner.lock().unwrap().sender.subscribe() };
+
+        loop {
+            loop {
+                std::thread::sleep(std::time::Duration::from_micros(10));
+
+                // Receive answer from the cameras
+                let (header, message) = match receiver.try_recv() {
+                    Ok(Message::ToBeSent(message)) => message,
+                    Err(broadcast::error::TryRecvError::Closed) => {
+                        unreachable!(
+                            "Closed channel: This should never happen, this channel is static!"
+                        );
+                    }
+                    Ok(Message::Received(_)) | Err(_) => continue,
```

Comment:
Maybe:
```suggestion
                    something => {
                        warning!("Got something unexpected on the channel: {something:#?}");
                        continue;
                    }
```

Reply:
If the channel is working properly, the `Message::Received(_)` is expected to happen for every received message because we are sharing a single channel to both send and receive from/to between mavlink and our cameras. The other error is when the channel is empty, so we want to ignore both.

I'll add a comment to state that.